File: bills.py
def separate_pellets(num_pellets):
    boxes = [1000000, 500000.0, 250000.0, 200000.0, 125000.0, 100000, 50000.0, 25000.0, 20000.0, 12500.0, 10000, 5000.0, 2500.0, 2000.0, 1250.0, 1000, 500.0, 250.0, 200.0, 125.0, 100, 50.0, 25.0, 20.0, 10, 5, 2, 1]
    boxes.sort()
    result = []
    for box in boxes:
        i=0
        print("BOX: ", box, num_pellets)
        while num_pellets > 0 and num_pellets >= box and i < 10:
            result.append(box)
            print(num_pellets)
            num_pellets -= box
            i+=1
    boxes.sort(reverse=True)
    while num_pellets > 0:
        for box in boxes:
            print("BOX: ", box, num_pellets)
            while num_pellets > 0 and num_pellets >= box:
                result.append(box)
                print(num_pellets)
                num_pellets -= box
    sum = 0
    for i in result:
        sum += i
    print("sum:", sum, len(result), boxes)
    return result
print(separate_pellets(100*(10**6)))

File: bills.c
#include <stdio.h>

int main() {
    const long bsize=28;
    long boxes[28] = {1000000, 500000, 250000, 200000, 125000, 100000, 50000, 25000, 20000, 12500, 10000, 5000, 2500, 2000, 1250, 1000, 500, 250, 200, 125, 100, 50, 25, 20, 10, 5, 2, 1};
    long num_pellets = 1234567;
    int i = 0, j;
    for (j = bsize-1; j >=0; j--) {
      i=0;
      while (num_pellets > 0 && num_pellets >= boxes[j] && i < 10) {
	if(num_pellets<boxes[j]) break;
	printf("%ld,", boxes[j]);
	fflush(stdout);
	num_pellets -= boxes[j];
	i++;
      }
    }
    for (j = 0; j < bsize && num_pellets>0; j++) {
      while (num_pellets > 0 && num_pellets >= boxes[j]) {
	printf("%ld,", boxes[j]);
	num_pellets -= boxes[j];
      }
    }
    printf("0\n");
    return 0;
}
